Provide more visibility into BPF filesystem mounting

tgraf opened the pull request:
Related: #2924

manalibhutiyani commented on pkg/bpf/bpffs.go:
```diff
@@ -210,12 +197,22 @@ func mountFS() error {
+	// into a container.
+	magic := uint32(0xCAFE4A11)
+	if uint32(fsdata.Type) != magic {
+		log.WithField(logfields.Path, mapRoot).Warning("BPF root is not a BPF filesystem (%d != %d)",
```
Should this be a `Log.Fatal` instead, now that we can detect that the mounted BPF file system is not of kind BPF?

manalibhutiyani replied:
Actually how is this different from the `IsBpffs` logic we had before? I am a bit confused?

tgraf replied:
It was previously casting to `int32` while `0xCAFE4A11` requires an `uint32`

```
-	magic := uint32(0xCAFE4A11)
-	var fsdata unix.Statfs_t
-	if err := unix.Statfs(path, &fsdata); err != nil {
-		log.WithField(logfields.Path, path).Error("BPF filesystem path is not mounted")
-		return false
-	}
-	return int32(magic) == int32(fsdata.Type)
```

I'll push another commit which changes this to `stat` and we can make it fail hard as it works in container environments.

tgraf committed:
bpf: Move pipe command helper to pkg/pipeexec

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
bpf: Print info message when BPF filesystem is mounted

Also warn if BPF root is not a BPF filesystem

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
cilium: Do not unmount BPF filesystem on cleanup

It does not seem wise to unmount the BPF filesystem on manual cleanup, it may
be in use by other components.

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
bpf: Wait with mounting delayed opens until filesystem is really mounted

The current code forced opening of delayed BPF map opens in SetMapRoot() which
is invoked before MountFS(). MountFS() is always called during init but after
SetMapRoot(). MountFS() will force the delayed opens. Leave it to MountFS() to
open all maps.

This fixes a bug when the path to the BPF filesystem was specified in the agent
but the path was not yet mounted.

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
bpf: Warn if BPF root is not a BPF filesystem

Also fixes a signed int32 as the BPF filesystem magic value must be an uint32

Signed-off-by: Thomas Graf <thomas@cilium.io>

joestringer reviewed (approved):
When it's volume-mapped into the container can we determine this based on the magic value and attempt to find the underlying FS?

Changes LGTM otherwise.
